src/sparseml/pytorch/image_classification/train.py

```diff
@@ -12,528 +12,613 @@
-    default_optim = "SGD"
-    optim: str = field(
-        default=default_optim,
-        metadata={
-            "help": f"The optimizer type to use, one of {optim_choices}."
-            f" Defaults to `{default_optim}`"
-        },
-    )
-    optim_args: json.loads = field(
-        default_factory=lambda: {
+@click.command()
+@click.option(
+    "--train-batch-size",
+    "--train_batch_size",
+    type=int,
+    required=True,
+    help="Train batch size",
+)
+@click.option(
+    "--test-batch-size",
+    "--test_batch_size",
+    type=int,
+    required=True,
+    help="Test/Validation batch size",
+)
+@click.option(
+    "--dataset",
+    "-d",
+    type=str,
+    required=True,
+    help="The dataset to use for training, "
+    "ex: `imagenet`, `imagenette`, `cifar10`, etc. "
+    "Set to `imagefolder` for a generic dataset setup with "
+    "imagefolder type structure like imagenet or loadable by "
+    "a dataset in `sparseml.pytorch.datasets`",
+)
+@click.option(
+    "--dataset-path",
+    "--dataset_path",
+    type=click.Path(dir_okay=True, file_okay=False),
+    callback=helpers.create_dir_callback,
+    required=True,
+    help="The root dir path where the dataset is stored or should "
+    "be downloaded to if available",
+)
+@click.option(
+    "--arch_key",
+    "--arch-key",
+    type=str,
+    default=None,
+    help="The architecture key for image classification model; "
+    "example: `resnet50`, `mobilenet`. "
+    "Note: Will be read from the checkpoint if not specified",
+)
+@click.option(
+    "--local_rank",
+    "--local-rank",
+    type=int,
+    default=-1,
+    help="Local rank for distributed training",
+)
+@click.option(
+    "--checkpoint-path",
+    "--checkpoint_path",
+    type=str,
+    default=None,
+    help="A path to a previous checkpoint to load the state from "
+    "and resume the state for. If provided, pretrained will "
+    "be ignored . If using a SparseZoo recipe, can also "
+    "provide 'zoo' to load the base weights associated with "
+    "that recipe. Additionally, can also provide a SparseZoo model stub "
+    "to load model weights from SparseZoo",
+)
+@click.option(
+    "--init-lr",
+    type=float,
+    default=1e-9,
+    show_default=True,
+    help="The initial learning rate to use while training, "
+    "the actual initial value used will be set by the"
+    "sparseml recipe",
+)
+@click.option(
+    "--recipe-path",
+    "--recipe_path",
+    type=str,
+    default=None,
+    help="The path to the yaml/md file containing the modifiers and "
+    "schedule to apply them with. Can also provide a "
+    "SparseZoo stub prefixed with 'zoo:' with an optional "
+    "'?recipe_type=' argument",
+)
+@click.option(
+    "--eval-mode/--no-eval-mode",
+    "--eval_mode/--no_eval_mode",
+    "--eval/--no-eval",
+    is_flag=True,
+    show_default=True,
+    help="Puts model into evaluation mode (Model weights are not updated)",
+)
+@click.option(
+    "--optim",
+    "--optimizer",
```

Comment:
let's not alias arguments here as it could lead to confusion on if they are different or same args